### ISSUE TITLE
org.bouncycastle.crypto.fips.FipsSelfTestFailedError: Exception on self test: Array index out of range: 32: EC

### ISSUE THREAD

**Issue by nmanda23** [bcgit/bc-java#1532]
getting the following stacktrace when upgraded bc-fips-1.0.2.3.jar to  bc-fips-1.0.2.4.jar. No issues when bc-fips-1.0.2.3.jar is used.  
This is with IBM Java 8.

 ERROR c.s.s.p.SecurityProperties - org.bouncycastle.crypto.fips.FipsSelfTestFailedError: Exception on self test: Array index out of range: 32: EC

              at org.bouncycastle.crypto.fips.SelfTestExecutor.validate(Unknown Source)

              at org.bouncycastle.crypto.fips.FipsEC.f2mDsaTest(Unknown Source)

              at org.bouncycastle.crypto.fips.FipsEC.<clinit>(Unknown Source)

              at java.lang.Class.forNameImpl(Native Method)

              at java.lang.Class.forName(Class.java:340)

              at org.bouncycastle.crypto.fips.FipsStatus.loadClass(Unknown Source)

              at org.bouncycastle.crypto.fips.FipsStatus.access$200(Unknown Source)

              at org.bouncycastle.crypto.fips.FipsStatus$Loader.run(Unknown Source)

              at org.bouncycastle.crypto.fips.FipsStatus.isReady(Unknown Source)

              at org.bouncycastle.crypto.CryptoServicesRegistrar.getDefaultMode(Unknown Source)

              at org.bouncycastle.crypto.CryptoServicesRegistrar.<clinit>(Unknown Source)

              at org.bouncycastle.jcajce.provider.ProvSecureHash$MD5.configure(Unknown Source)

              at org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider.<init>(Unknown Source)

              at org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider.<init>(Unknown Source)

              at org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider.<init>(Unknown Source)

              at com.sterlingcommerce.hadrian.common.util.BCSecurityFipsUtilsImpl.<clinit>(BCSecurityFipsUtilsImpl.java:126)

              at java.lang.Class.forNameImpl(Native Method)

              at java.lang.Class.forName(Class.java:340)

              at com.sterlingcommerce.security.provider.SecurityProperties.getBCSecurityUtils(SecurityProperties.java:465)

              at com.sterlingcommerce.security.provider.SecurityProperties.initialize(SecurityProperties.java:705)

              at com.sterlingcommerce.security.provider.SecurityProperties.load(SecurityProperties.java:224)

              at com.sterlingcommerce.security.provider.SecurityProperties.load(SecurityProperties.java:158)

              at com.sterlingcommerce.hadrian.Main.isBootStrapEnabled(Main.java:762)

              at com.sterlingcommerce.hadrian.Main.main(Main.java:279)

**Comment by nmanda23** [comment]
Including email from David Hook on this issue:
================================

**Comment by dghgit** [comment]
This is now fixed in the early access version. Still in negotiations about organizing a release date.

**Comment by nmanda23** [comment]
Thank you David.

Naru

**Comment by dghgit** [comment]
If you've got a support contract 1.0.2.5 is now in early access, otherwise you'll need to role back to 1.0.2.3.

**Comment by smartycardpants** [comment]
I downloaded 1.0.2.3 and I no longer get the FipsSelfTestFailedError. However, I noticed that there is a vulnerability in that jar:  CVE-2022-45146. Has this vulnerability been fixed? 

Thanks,
smartycardpants

**Comment by dghgit** [comment]
Yes, in 1.0.2.4

**Comment by smartycardpants** [comment]
Yes, I guessed it was fixed in 1.0.2.4, but that's where I ran into the FipsSelfTestFailedError which I cannot get around.

**Comment by smartycardpants** [comment]
Thank you, David.

Another question - 1.0.2.4 has the FipsSelfTestFailedError that you mentioned is fixed in the early access version (1.0.2.5). When will 1.0.2.5 be publicly available without a support contract? Also, what does a support contract entail - cost? duration? etc. (Thanks again.)

**Comment by dghgit** [comment]
Details about support and early access are available here https://www.keyfactor.com/open-source/bouncy-castle-support/ Contracts start at 25K USD. I'd guess we're currently looking at about 6 months for public release - the changes we are making are small so we're not anticipating any issues there, however the CMVP is under a lot of strain at the moment.

**Comment by dghgit** [comment]
1.0.2.5 and 1.0.2.6 are now publicly available.
